### PR TITLE
[plugin.video.srf_ch_replay] 2.0.0

### PR DESCRIPTION
**Pull request by ManBehindMooN**
### Description
New Plugin: SRF Replay plugin lists and plays all recorded shows from the national Swiss German TV station SRF. All the shows are mainly in German.
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.srf_ch_replay/addon.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+#'Base settings'
+#'Start of the plugin functionality is at the end of the file'
+addon = xbmcaddon.Addon()
+addonID = 'plugin.video.srf_podcast_ch'
+pluginhandle = int(sys.argv[1])
+socket.setdefaulttimeout(30)
+xbmcplugin.setPluginCategory(pluginhandle, "News")
+xbmcplugin.setContent(pluginhandle, "tvshows")
+addon_work_folder = xbmc.translatePath("special://profile/addon_data/" + addonID)
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.py]: xbmcvfs
```suggestion
addon_work_folder = xbmcvfs.translatePath("special://profile/addon_data/" + addonID)
```

File: plugin.video.srf_ch_replay/addon.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+if not os.path.isdir(addon_work_folder):
+    os.mkdir(addon_work_folder)
+FavoritesFile = xbmc.translatePath("special://profile/addon_data/" + addonID + "/" + addonID + ".favorites")
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.py]:
```suggestion
FavoritesFile = xbmcvfs.translatePath("special://profile/addon_data/" + addonID + "/" + addonID + ".favorites")
```

File: plugin.video.srf_ch_replay/addon.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+forceViewMode = addon.getSetting("forceView") == "true"
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.py]: Not allowed

File: plugin.video.srf_ch_replay/addon.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    request.add_header('Accept-encoding', 'gzip')
+    response = ''
+    try:
+        response = urllib.request.urlopen(urlstring)
+        if response.info().get('Content-Encoding') == 'gzip':
+            buf = StringIO(response.read())
+            f = gzip.GzipFile(fileobj=buf)
+            response = StringIO(f.read())
+    except Exception as e:
+        print(traceback.format_exc())
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.py]: no prints please, use xbmc.log

File: plugin.video.srf_ch_replay/addon.py [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        xbmcgui.Dialog().ok('xStream Error', str(e.__class__.__name__), str(e))
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.py]: Any string shown on the interface has to be localized

File: plugin.video.srf_ch_replay/addon.py [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+            desc = 'No Description'
+        try:
+            picture = show['Image']['ImageRepresentations']['ImageRepresentation'][0]['url']
+        except:
+            picture = ''
+        _addShow(title, show['id'], mode, desc, picture, page)
+
+    xbmcplugin.addSortMethod(pluginhandle, 1)
+    xbmcplugin.endOfDirectory(pluginhandle)
+    if forceViewMode:
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.py]: not allowed, please remove

File: plugin.video.srf_ch_replay/addon.py [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+        title = episode['title']
+        url = ''
+        desc = ''
+        picture = ''
+        pubdate = episode['publishedDate']
+
+        try:
+            desc = episode['Assets']['Video'][0]['AssetMetadatas']['AssetMetadata'][0]['description']
+        except:
+            desc = 'No Description'
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.py]: This has to be localized

File: plugin.video.srf_ch_replay/addon.py [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+
+    # check if another page is available
+    page = int(page)
+    maxpage = int(maxpage)
+    if page < maxpage or maxpage == 0 and len(show) == int(numberOfEpisodesPerPage):
+        page = page + 1
+        _addnextpage(tr(30005).format(page, maxpage), showid, 'listEpisodes', '', showbackground, page)
+
+    xbmcplugin.endOfDirectory(pluginhandle)
+    if forceViewMode:
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.py]: please remove

File: plugin.video.srf_ch_replay/addon.xml [hunk 9]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<addon id="plugin.video.srf_ch_replay" name="SRF Replay" version="2.0.0" provider-name="Man Behind MooN">
+  <requires>
+    <import addon="xbmc.python" version="3.0.0"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource" library="addon.py">
+    <provides>video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+	 <news>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]: either add news or remove the element

File: plugin.video.srf_ch_replay/addon.xml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+	<assets>
+		<icon>icon.png</icon>
+	</assets>
+    <summary lang="en_gb">SRF Replay plugin lists all shows from SRF.</summary>
+    <description lang="en_gb">SRF Replay plugin lists all recorded shows from the national Swiss German TV station SRF. All the shows are mainly in German.</description>
+    <disclaimer lang="en_gb">This plugin is not endorsed by SRF</disclaimer>
+	<summary lang="de_de">SRF Replay Plugin zeigt alle SRF Sendungen.</summary>
+    <description lang="de_de">Der SRF Replay Plugin zeigt alle aufgezeichneten Sendungen vom nationalen Schweizer deutschprachigen TV Sender SRF. Die Sendungen sind hauptsächlich in deutscher Spracher.</description>
+    <disclaimer lang="de_de">Dieser Plugin ist nicht offiziell von SRF</disclaimer>
+    <language>de_de</language>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]: de
```suggestion
    <language>de</language>
```

File: plugin.video.srf_ch_replay/addon.xml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    <platform>all</platform>
+    <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]:
```suggestion
    <license>GPL-2.0-only</license>
```
or
```suggestion
    <license>GPL-2.0-or-later</license>
```
SPDX identifier

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments, thanks for the submission

File: plugin.video.srf_ch_replay/addon.py (hunk 4 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.py]: done

File: plugin.video.srf_ch_replay/addon.py (hunk 5 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.py]: everything should be localized now

File: plugin.video.srf_ch_replay/addon.py (hunk 7 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.py]: done

File: plugin.video.srf_ch_replay/addon.py (hunk 8 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.py]: removed

File: plugin.video.srf_ch_replay/addon.xml (hunk 9 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: added news

File: plugin.video.srf_ch_replay/addon.xml (hunk 10 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: changed to de

File: plugin.video.srf_ch_replay/addon.xml (hunk 11 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: changed to <license>GPL-2.0-or-later</license>

**Comment by ManBehindMooN** [comment]
@enen92: First of all thank you for the thorough review and I appreciated it. I went through the whole source code again and changed everything according to your suggestions. I hope it is all good now. Otherwise just drop me a line and I will fix it asap. Thank you.

File: plugin.video.srf_ch_replay/addon.py (hunk 6 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.py]: removed

File: plugin.video.srf_ch_replay/addon.py (hunk 2 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.py]: changed

File: plugin.video.srf_ch_replay/addon.xml [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    <summary lang="en_gb">SRF Replay plugin lists all shows from SRF.</summary>
+    <description lang="en_gb">SRF Replay plugin lists all recorded shows from the national Swiss German TV station SRF. All the shows are mainly in German.</description>
+    <disclaimer lang="en_gb">This plugin is not endorsed by SRF</disclaimer>
+	<summary lang="de_de">SRF Replay Plugin zeigt alle SRF Sendungen.</summary>
+    <description lang="de_de">Der SRF Replay Plugin zeigt alle aufgezeichneten Sendungen vom nationalen Schweizer deutschprachigen TV Sender SRF. Die Sendungen sind hauptsächlich in deutscher Spracher.</description>
+    <disclaimer lang="de_de">Dieser Plugin ist nicht offiziell von SRF</disclaimer>
+    <language>de</language>
+    <platform>all</platform>
+    <license>GPL-2.0-or-later</license>
+    <forum></forum>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]: Sorry I did not spot this earlier, please drop if empty or fill with a support thread

File: plugin.video.srf_ch_replay/addon.xml [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    <website></website>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]: Same for website

File: plugin.video.srf_ch_replay/addon.xml [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+  </requires>
+  <extension point="xbmc.python.pluginsource" library="addon.py">
+    <provides>video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+	 <news>Former "SRF Podcast Plugin" reloaded as "SRF Replay" for Kodi 19</news>
+	<assets>
+		<icon>icon.png</icon>
+	</assets>
+    <summary lang="en_gb">SRF Replay plugin lists all shows from SRF.</summary>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]:
```suggestion
    <summary lang="en_GB">SRF Replay plugin lists all shows from SRF.</summary>
```

File: plugin.video.srf_ch_replay/addon.xml [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+    <description lang="en_gb">SRF Replay plugin lists all recorded shows from the national Swiss German TV station SRF. All the shows are mainly in German.</description>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]:
```suggestion
    <description lang="en_GB">SRF Replay plugin lists all recorded shows from the national Swiss German TV station SRF. All the shows are mainly in German.</description>
```

File: plugin.video.srf_ch_replay/addon.xml [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+    <disclaimer lang="en_gb">This plugin is not endorsed by SRF</disclaimer>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]:
```suggestion
    <disclaimer lang="en_GB">This plugin is not endorsed by SRF</disclaimer>
```

File: plugin.video.srf_ch_replay/addon.xml [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+	<summary lang="de_de">SRF Replay Plugin zeigt alle SRF Sendungen.</summary>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]:
```suggestion
	<summary lang="de_DE">SRF Replay Plugin zeigt alle SRF Sendungen.</summary>
```

File: plugin.video.srf_ch_replay/addon.xml [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+    <description lang="de_de">Der SRF Replay Plugin zeigt alle aufgezeichneten Sendungen vom nationalen Schweizer deutschprachigen TV Sender SRF. Die Sendungen sind hauptsächlich in deutscher Spracher.</description>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]:
```suggestion
    <description lang="de_DE">Der SRF Replay Plugin zeigt alle aufgezeichneten Sendungen vom nationalen Schweizer deutschprachigen TV Sender SRF. Die Sendungen sind hauptsächlich in deutscher Spracher.</description>
```

File: plugin.video.srf_ch_replay/addon.xml [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+    <disclaimer lang="de_de">Dieser Plugin ist nicht offiziell von SRF</disclaimer>
```
Comment (enen92) [inline plugin.video.srf_ch_replay/addon.xml]:
```suggestion
    <disclaimer lang="de_DE">Dieser Plugin ist nicht offiziell von SRF</disclaimer>
```

**Review by enen92 — CHANGES_REQUESTED** [review]

**Comment by enen92** [comment]
Travis is still failing to pass (see the language requests I did above)

File: plugin.video.srf_ch_replay/addon.xml (hunk 12 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: no problem... done

File: plugin.video.srf_ch_replay/addon.xml (hunk 13 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: done

File: plugin.video.srf_ch_replay/addon.xml (hunk 14 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: done

File: plugin.video.srf_ch_replay/addon.xml (hunk 15 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: done

File: plugin.video.srf_ch_replay/addon.xml (hunk 16 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: done

File: plugin.video.srf_ch_replay/addon.xml (hunk 17 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: done

File: plugin.video.srf_ch_replay/addon.xml (hunk 18 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: done

File: plugin.video.srf_ch_replay/addon.xml (hunk 19 above)
Comment (ManBehindMooN) [reply plugin.video.srf_ch_replay/addon.xml]: done

**Review by enen92 — APPROVED** [review]
Thanks very much for adjusting

**Pull request merged**